pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountThetaSketchAggregationFunction.java

```diff
@@ -526,28 +543,51 @@ private Predicate getPredicate(String predicateString) {
    * @param sketchMap Precomputed sketches for predicates that are part of the expression.
    * @return Overall evaluated sketch for the expression.
    */
-  private Sketch evalPostAggregationExpression(FilterContext postAggregationExpression,
-      Map<Predicate, Sketch> sketchMap) {
-    switch (postAggregationExpression.getType()) {
-      case AND:
-        Intersection intersection = _setOperationBuilder.buildIntersection();
-        for (FilterContext child : postAggregationExpression.getChildren()) {
-          intersection.update(evalPostAggregationExpression(child, sketchMap));
-        }
-        return intersection.getResult();
-      case OR:
-        Union union = _setOperationBuilder.buildUnion();
-        for (FilterContext child : postAggregationExpression.getChildren()) {
-          union.update(evalPostAggregationExpression(child, sketchMap));
+  private Sketch evalPostAggregationExpression(
+      final ExpressionContext postAggregationExpression,
+      final List<Predicate> expressions,
+      final Map<Predicate, Sketch> sketchMap) {
+    if (postAggregationExpression.getType() == ExpressionContext.Type.LITERAL) {
+      throw new IllegalArgumentException("Literal not supported in post-aggregation function");
+    }
+
+    if (postAggregationExpression.getType() == ExpressionContext.Type.IDENTIFIER) {
+      final Predicate exp =
+          expressions.get(extractSubstitutionPosition(postAggregationExpression.getLiteral()) - 1);
+      return sketchMap.get(exp);
+    }
+
+    // shouldn't throw exception because of the validation in the constructor
+    final MergeFunction func =
+        MergeFunction.valueOf(postAggregationExpression.getFunction().getFunctionName().toUpperCase());
+
+    // handle functions recursively
+    switch(func) {
+      case SET_UNION:
+        final Union union = _setOperationBuilder.buildUnion();
+        for (final ExpressionContext exp : postAggregationExpression.getFunction().getArguments()) {
+          union.update(evalPostAggregationExpression(exp, expressions, sketchMap));
         }
         return union.getResult();
-      case PREDICATE:
-        return sketchMap.get(postAggregationExpression.getPredicate());
+      case SET_INTERSECT:
+        final Intersection intersection = _setOperationBuilder.buildIntersection();
+        for (final ExpressionContext exp : postAggregationExpression.getFunction().getArguments()) {
+          intersection.update(evalPostAggregationExpression(exp, expressions, sketchMap));
+        }
+        return intersection.getResult();
+      case SET_DIFF:
+        final List<ExpressionContext> args = postAggregationExpression.getFunction().getArguments();
+        final AnotB diff = _setOperationBuilder.buildANotB();
+        final Sketch a = evalPostAggregationExpression(args.get(0), expressions, sketchMap);
+        final Sketch b = evalPostAggregationExpression(args.get(1), expressions, sketchMap);
+        diff.update(a, b);
```

Comment:
Lets make sure we call out in the documentation that the order matters in case of SET_DIFF

Reply:
Good call. Done - updated in the release notes in the PR.